src/connect4.ts

```diff
@@ -1,95 +1,24 @@
-import * as readline from "readline";
-
-export enum boardLayout {
-  PLACEHOLDER = "$",
-  TOP = "+---+---+---+---+---+---+---+",
-  BOTTOM = "+---+---+---+---+---+---+---+\n  1   2   3   4   5   6   7  ",
-  INFORMATIONS = "\n-- You are player: " + PLACEHOLDER + " --",
-  PROMPT = "Please enter the column number you want to play your token $ [1-7]: ",
-  ERROR_INVALID_COLUMN_NUMBER = "Invalid column number.\n",
-  ERROR_COLUMN_FULL = "Cannot add token into column $ which is full.\n",
-  SEPARATOR = "|",
-  PLAYER_ONE_TOKEN = "o",
-  PLAYER_TWO_TOKEN = "x",
-  EMPTY_TOKEN = " ",
-  NB_ROWS = 6,
-  NB_COLUMN = 7,
-}
-
-export enum PlayerNum {
-  p1 = 1,
-  p2 = 2,
-  empty = 0,
-}
-export type BoardState = Array<Array<PlayerNum>>;
-
-export interface GameState {
-  boardState: BoardState;
-  currentPlayer: PlayerNum;
-}
-
-export function getPlayerTokenChar(playerNum: PlayerNum) {
-  switch (playerNum) {
-    case PlayerNum.p1:
-      return boardLayout.PLAYER_ONE_TOKEN;
-    case PlayerNum.p2:
-      return boardLayout.PLAYER_TWO_TOKEN;
-    case PlayerNum.empty:
-      return boardLayout.EMPTY_TOKEN;
-  }
-}
-
-export function printBoardStateToConsole(boardState: String): void {
-  console.clear();
-  console.log(boardState);
-}
-
-export function boardStateToString(gameState: GameState): String {
-  const resultDisplay: Array<String> = [];
-
-  resultDisplay.push(`\n${boardLayout.TOP}\n`);
-
-  gameState.boardState.forEach((line: Array<number>) => {
-    line.forEach((token: number) => {
-      resultDisplay.push(
-        `${boardLayout.SEPARATOR} ${
-          token === 1
-            ? boardLayout.PLAYER_ONE_TOKEN
-            : token === 2
-              ? boardLayout.PLAYER_TWO_TOKEN
-              : boardLayout.EMPTY_TOKEN
-        } `,
-      );
-    });
-    resultDisplay.push(`${boardLayout.SEPARATOR}\n`);
-  });
-
-  resultDisplay.push(`${boardLayout.BOTTOM}\n`);
-
-  // Player information
-  const p1Token = getPlayerTokenChar(PlayerNum.p1);
-  resultDisplay.push(
-    `${boardLayout.INFORMATIONS.replace(boardLayout.PLACEHOLDER, p1Token)}\n`,
-  );
-
-  return resultDisplay.join("");
-}
+import { messages, PLACEHOLDER } from "./config/messages.js";
+import { BoardState, GameState, PlayerNum } from "./types/gameState.js";
+import {
+  boardLayout,
+  boardStateToString,
+  printBoardStateToConsole,
+} from "./layout/cliLayout.js";
+import { readNextPlay } from "./prompt.js";
+import { transpose } from "./tools.js";
 
 /**
  * @throws Error if the played column is already full
  */
-export function playToken(
-  boardState: BoardState,
-  column: number,
-  pNum: number,
-): BoardState {
-  if (column < 0 || column >= boardLayout.NB_COLUMN) {
-    throw new Error(boardLayout.ERROR_INVALID_COLUMN_NUMBER);
+export function playToken(gameState: GameState, column: number): GameState {
+  if (column < 0 || column > boardLayout.NB_COLUMN) {
```

Comment:
```suggestion
  if (column <= 0 || column > boardLayout.NB_COLUMN) {
```
As the first column = 1 and not 0